tools/scripts/uconfig_vars_test.py

```diff
@@ -0,0 +1,217 @@
+      command: string with the command.
+
+    Returns: stdout and exit code of command execution.
+    """
+
+    command += ' >> uconfig_test.log 2>&1'
+    p = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE,
+                         stderr=subprocess.STDOUT, close_fds=True)
+    stdout, _ = p.communicate()
+    return stdout, p.returncode
+
+def SetUpICU():
+    """Configuration, installation of ICU4C."""
+
+    RunCmd('mkdir /tmp/icu_cnfg')
+    out, exit_code = RunCmd('./runConfigureICU Linux --prefix=/tmp/icu_cnfg')
+    if exit_code != 0:
+        print('ICU4C configuration failed!')
+        print(out)
+        sys.exit(-1)
+    _, exit_code = RunCmd('make -j2 install')
+    if exit_code != 0:
+        print('make install failed!')
+        sys.exit(-1)
+    # The header test is set up in a way that it uses uconfig.h in /tmp/icu_cnfg
+    # while the ICU4C unit tests uses common/unicode/uconfig.h. Set a link.
+    RunCmd('rm /tmp/icu_cnfg/include/unicode/uconfig.h')
+    os.symlink(os.path.abspath('common/unicode/uconfig.h'),
+               '/tmp/icu_cnfg/include/unicode/uconfig.h')
+    print('ICU setup completed.')
+
+def ExtractUConfigNoXXX(uconfig_file):
+    """Parses uconfig.h and returns a list of UCONFIG_NO_XXX labels.
+    Initializes test result structure.
+
+    Args:
+      uconfig_file: common/unicode/uconfig.h as string.
+
+    Returns:
+      List of all UCONFIG_NO_XXX flags found in uconfig.h, initialized test
+      result structure.
+    """
+
+    uconfig_no_flags_all = []
+    test_results = {}
+    uconfig_no_regex = r'UCONFIG_NO_[A-Z_]*'
+
+    for uconfig_no_flag in re.finditer(uconfig_no_regex, uconfig_file):
+        if uconfig_no_flag.group(0) not in uconfig_no_flags_all:
+            uconfig_no_flags_all.append(uconfig_no_flag.group(0))
+
+    test_results = {f: {'unit_test': False, 'hdr_test': False} for f in uconfig_no_flags_all}
+    test_results['all_flags'] = {'unit_test': False, 'hdr_test' : False}
+
+    return uconfig_no_flags_all, test_results
+
+def EnableUConfigNo(uconfig_file, uconfig_no_flag):
+    """Sets the uconfig_no_flag in common/unicode/uconfig.h file.
+
+    Args:
+      uconfig_file: common/unicode/uconfig.h as string.
+      uconfig_no_flag: flags that is to be set to 1.
+
+    Returns:
+      uconfig.h with flag set to 1
+    """
+
+    uconfig_no_def_regex = r'(?m)#ifndef %s\n#\s+define %s\s+0\n#endif$' % (
+        uconfig_no_flag, uconfig_no_flag)
+    uconfig_no_def_match = re.search(uconfig_no_def_regex, uconfig_file)
+    if not uconfig_no_def_match:
+        print('No definition for flag %s found!\n' % uconfig_no_flag)
+        sys.exit(-1)
+    uconfig_no_def = uconfig_no_def_match.group(0)
+    set_uconfig_no_def = uconfig_no_def.replace(' 0\n', ' 1\n')
+    set_in_uconfig_file = uconfig_file.replace(uconfig_no_def, set_uconfig_no_def)
+
+    return set_in_uconfig_file
+
+def main():
+    # Read the options and determine what to run.
+    run_hdr = False
```

Comment:
optional: If we wanted to reduce the runtime further, we could have more actions that each test fewer combinations. This might mean hardcoding the list of options in the form of one GHA each (and some test to make sure we don't miss new options). So... probably not now. :-)